jujugui/static/gui/src/app/components/user-profile/entity/entity.js

```diff
@@ -358,6 +468,9 @@ YUI.add('user-profile-entity', function() {
         });
         destroyButton = this._generateDestroyButton(entity);
       } else {
+        if (isCharm) {
```

Comment:
I feel like this should be done in `componentWillMount` as that gives it a bit of time to go fetch the content before it renders. And you don't need to fetch the metrics every time it renders...